tests/e2e/common.go

```diff
@@ -215,10 +244,27 @@ func (td *OsmTestData) GetOSMInstallOpts() InstallOSMOpts {
 	}
 }
 
-// InstallOSM installs OSM. Right now relies on externally calling the binary and a subset of possible opts
-// TODO: refactor install to be able to call it directly here vs. exec-ing CLI.
+// InstallOSM installs OSM. The behavior of this function is dependant on
+// installType and instOpts
 func (td *OsmTestData) InstallOSM(instOpts InstallOSMOpts) error {
-	if td.kindCluster {
+	if td.instType == NoInstall {
+		if instOpts.certManager != defaultCertManager ||
+			instOpts.deployPrometheus ||
+			instOpts.deployGrafana ||
+			instOpts.deployJaeger {
```

Comment:
Should we explicitly define defaults for these values so there isn't a hidden assumption here that the default is false?

Reply:
+1, will do